packages/react/kibana_context/root/root_provider.tsx

```diff
@@ -25,6 +27,8 @@ export interface KibanaRootContextProviderProps extends KibanaEuiProviderProps {
   i18n: I18nStart;
   /** The `AnalyticsServiceStart` API from `CoreStart`. */
   analytics?: Pick<AnalyticsServiceStart, 'reportEvent'>;
+  /** The `ExecutionContextStart` API from `CoreStart`. */
+  executionContext?: ExecutionContextStart;
```

Comment:
Q: to avoid errors in the future, should we make this one mandatory?

Reply:
I will check to see if it is OK to make this dependency mandatory.
Still, I don't think this will prevent similar issues in the future because, eventually, it will be passed to SharedUXContext, which is initialized with an empty executionContext [here](https://github.com/elastic/kibana/blob/main/packages/shared-ux/router/impl/services.ts#L67).